src/components/FeatureViews/Courses/CourseManagementContainer.js

```diff
@@ -1,463 +1,638 @@
-        checkFilter(course.courseCategory.id, subectFilterValue) &&
-        checkFilter(course.instructor.user.id, instructorsFilterValue)
-    )
-    .sort(
-      (firstEl, secondEl) =>
-        ({
-          start_date: sortDescOrder(firstEl.startDate, secondEl.startDate),
-          class_name: sortDescOrder(firstEl.title, secondEl.title),
-        }[sortByDate])
+    const {
+        data: studentData,
+        loading: studentLoading,
+        error: studentError,
+    } = useQuery(GET_STUDENTS, {
+        variables: { accountId },
+        skip: accountInfo.accountType !== 'PARENT',
+    });
+
+    if (courseLoading || studentLoading) return <Loading />;
+    if (courseError) return console.error(courseError.message);
+    if (studentError) return console.error(studentError.message);
+
+    const createFilteredListFromCourses = (filterCondition) =>
+        courseData.courses.reduce(
+            (accumulator, currentValue) =>
+                !accumulator.some((course) =>
+                    filterCondition(currentValue, course)
+                )
+                    ? [...accumulator, currentValue]
+                    : accumulator,
+            []
+        );
+    const subjectList = createFilteredListFromCourses(
+        (currentValue, course) =>
+            currentValue.courseCategory.id === course.courseCategory.id
+    );
+    const instructorsList = createFilteredListFromCourses(
+        (currentValue, course) =>
+            currentValue.instructor.user.id === course.instructor.user.id
     );
 
-  return (
-    <Grid item xs={12}>
-      <BackgroundPaper elevation={2}>
-        <Typography align="left" className="heading" variant="h3">
-          Course Management
-        </Typography>
-        <Paper elevation={4} className={classes.appBar}>
-          <Grid
-            container
-            alignItems="center"
-            className={classes.containerMargins}
-          >
-            <Grid item xs={3}>
-              <FormControl className={classes.margin}>
-                <Select
-                  labelId="course-management-sort-tab"
-                  id="course-management-sort-tab"
-                  displayEmpty
-                  value={sortByDate}
-                  onChange={handleChange}
-                  classes={{ select: classes.menuSelect }}
-                  input={<BootstrapInput />}
-                  MenuProps={{
-                    classes: { list: classes.dropdown },
-                    anchorOrigin: {
-                      vertical: "bottom",
-                      horizontal: "left",
-                    },
-                    transformOrigin: {
-                      vertical: "top",
-                      horizontal: "left",
-                    },
-                    getContentAnchorEl: null,
-                  }}
+
+    const checkFilter = (value, filter) => '' === filter || value === filter;
+    const checkStudentEnrolled = (value, filter) =>
```

Comment:
I'm not sure how this "checkStudentEnrolled" this also looks very similar to "checkFilter" from earlier, why do you need this different function?

Reply:
@andrewjknapp wrote this out, but I think its because the ```checkFilter``` returns true or false when you only filter one element. ```checkStudentEnrolled``` find the students in an array and returns true if they are in that array. I guess the only change is that we don't need JSON.parse because we are already storing ```studentFilterValue```  as an array.

Reply:
still doesn't make sense to me but this is a problem for another time